### ISSUE TITLE
autoawq 0.2.6 causing conflicting dependencies

### ISSUE THREAD

**Issue by Mr-Jeffery** [axolotl-ai-cloud/axolotl#1819]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

I created a conda envirnoment with python==3.11 and torch==2.1.1+cu118 and conduct installation as readme guided, but the conda shows dependency conflicts.

### Current behaviour

autoawq 0.2.6 depends on torch==2.3.1, thus can not work with torch==2.1.1
```
ERROR: Cannot install None, axolotl and axolotl[deepspeed,flash-attn]==0.4.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    axolotl 0.4.1 depends on torch==2.1.1+cu118
    accelerate 0.32.0 depends on torch>=1.10.0
    bitsandbytes 0.43.3 depends on torch
    optimum 1.16.2 depends on torch>=1.11
    peft 0.12.0 depends on torch>=1.13.0
    trl 0.9.6 depends on torch>=1.4.0
    axolotl[deepspeed,flash-attn] 0.4.1 depends on torch==2.1.1+cu118
    deepspeed 0.14.4 depends on torch
    flash-attn 2.6.2 depends on torch
    autoawq 0.2.6 depends on torch==2.3.1
```

### Steps to reproduce

```
conda create -n axolotl python==3.11
conda activate axototl
pip3 install torch==2.1.1+cu118 --extra-index-url https://download.pytorch.org/whl/cu118 --no-cache-dir

git clone https://github.com/axolotl-ai-cloud/axolotl
cd axolotl

pip3 install packaging ninja
pip3 install -e '.[flash-attn,deepspeed]'
```

### Possible solution

We might need to change the `autoawq>=0.2.5` to `autoawq==0.2.5`

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

main/70978467a088da3abf3fe45d92d90f6529f19ea9

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by Mr-Jeffery** [comment]
`xformers==0.0.27` requires `torch==2.3.1` as well, I change to `torch==2.3.1+cu118` and installed the environment successfully, but the corresponding `flash_attn` would require `libcudart.so.12` from cuda 12+, I am not sure how to change my version now

**Comment by srushti98** [comment]
@Mr-Jeffery @winglian Is this resolved yet?
I am still getting this issue.

**Comment by Mr-Jeffery** [comment]
my problem is solved by changing the python version to 3.10 and with underlying packages:
```sh
$ conda list
# packages in environment at miniconda3/envs/axo:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
absl-py                   2.1.0                    pypi_0    pypi
accelerate                0.32.0                   pypi_0    pypi
addict                    2.4.0                    pypi_0    pypi
aiobotocore               2.13.1                   pypi_0    pypi
aiofiles                  23.2.1                   pypi_0    pypi
aiohappyeyeballs          2.3.5                    pypi_0    pypi
aiohttp                   3.10.2                   pypi_0    pypi
aioitertools              0.11.0                   pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
altair                    5.3.0                    pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
anyio                     4.4.0                    pypi_0    pypi
art                       6.2                      pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     24.2.0                   pypi_0    pypi
autoawq                   0.2.6                    pypi_0    pypi
autoawq-kernels           0.0.7                    pypi_0    pypi
axolotl                   0.4.1                     dev_0    <develop>
bitsandbytes              0.43.3                   pypi_0    pypi
botocore                  1.34.131                 pypi_0    pypi
bzip2                     1.0.8                h5eee18b_6    defaults
ca-certificates           2024.7.2             h06a4308_0    defaults
cachetools                5.4.0                    pypi_0    pypi
certifi                   2024.7.4                 pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
cmake                     3.30.2                   pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
coloredlogs               15.0.1                   pypi_0    pypi
contourpy                 1.2.1                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
datasets                  2.19.1                   pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
deepspeed                 0.14.4                   pypi_0    pypi
deepspeed-kernels         0.0.1.dev1698255861          pypi_0    pypi
dill                      0.3.8                    pypi_0    pypi
docker-pycreds            0.4.0                    pypi_0    pypi
docstring-parser          0.16                     pypi_0    pypi
einops                    0.8.0                    pypi_0    pypi
evaluate                  0.4.1                    pypi_0    pypi
exceptiongroup            1.2.2                    pypi_0    pypi
fastapi                   0.112.0                  pypi_0    pypi
fastcore                  1.6.3                    pypi_0    pypi
ffmpy                     0.4.0                    pypi_0    pypi
filelock                  3.15.4                   pypi_0    pypi
fire                      0.6.0                    pypi_0    pypi
flash-attn                2.6.2                    pypi_0    pypi
fonttools                 4.53.1                   pypi_0    pypi
frozenlist                1.4.1                    pypi_0    pypi
fschat                    0.2.36                   pypi_0    pypi
fsspec                    2024.3.1                 pypi_0    pypi
gcsfs                     2024.3.1                 pypi_0    pypi
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.43                   pypi_0    pypi
google-api-core           2.19.1                   pypi_0    pypi
google-auth               2.33.0                   pypi_0    pypi
google-auth-oauthlib      1.2.1                    pypi_0    pypi
google-cloud-core         2.4.1                    pypi_0    pypi
google-cloud-storage      2.18.1                   pypi_0    pypi
google-crc32c             1.5.0                    pypi_0    pypi
google-resumable-media    2.7.2                    pypi_0    pypi
googleapis-common-protos  1.63.2                   pypi_0    pypi
gradio                    3.50.2                   pypi_0    pypi
gradio-client             0.6.1                    pypi_0    pypi
grpcio                    1.65.4                   pypi_0    pypi
h11                       0.14.0                   pypi_0    pypi
hf-transfer               0.1.8                    pypi_0    pypi
hjson                     3.1.0                    pypi_0    pypi
httpcore                  1.0.5                    pypi_0    pypi
httpx                     0.27.0                   pypi_0    pypi
huggingface-hub           0.24.5                   pypi_0    pypi
humanfriendly             10.0                     pypi_0    pypi
idna                      3.7                      pypi_0    pypi
importlib-resources       6.4.0                    pypi_0    pypi
jinja2                    3.1.4                    pypi_0    pypi
jmespath                  1.0.1                    pypi_0    pypi
joblib                    1.4.2                    pypi_0    pypi
jsonschema                4.23.0                   pypi_0    pypi
jsonschema-specifications 2023.12.1                pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
latex2mathml              3.77.0                   pypi_0    pypi
ld_impl_linux-64          2.38                 h1181459_1    defaults
libffi                    3.3                  he6710b0_2    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
llvmlite                  0.43.0                   pypi_0    pypi
markdown                  3.6                      pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markdown2                 2.5.0                    pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
matplotlib                3.9.1.post1              pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
multidict                 6.0.5                    pypi_0    pypi
multiprocess              0.70.16                  pypi_0    pypi
ncurses                   6.4                  h6a678d5_0    defaults
networkx                  3.3                      pypi_0    pypi
nh3                       0.2.18                   pypi_0    pypi
ninja                     1.11.1.1                 pypi_0    pypi
numba                     0.60.0                   pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
nvidia-cublas-cu11        11.11.3.6                pypi_0    pypi
nvidia-cublas-cu12        12.1.3.1                 pypi_0    pypi
nvidia-cuda-cupti-cu11    11.8.87                  pypi_0    pypi
nvidia-cuda-cupti-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.8.89                  pypi_0    pypi
nvidia-cuda-nvrtc-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-runtime-cu11  11.8.89                  pypi_0    pypi
nvidia-cuda-runtime-cu12  12.1.105                 pypi_0    pypi
nvidia-cudnn-cu11         8.7.0.84                 pypi_0    pypi
nvidia-cudnn-cu12         8.9.2.26                 pypi_0    pypi
nvidia-cufft-cu11         10.9.0.58                pypi_0    pypi
nvidia-cufft-cu12         11.0.2.54                pypi_0    pypi
nvidia-curand-cu11        10.3.0.86                pypi_0    pypi
nvidia-curand-cu12        10.3.2.106               pypi_0    pypi
nvidia-cusolver-cu11      11.4.1.48                pypi_0    pypi
nvidia-cusolver-cu12      11.4.5.107               pypi_0    pypi
nvidia-cusparse-cu11      11.7.5.86                pypi_0    pypi
nvidia-cusparse-cu12      12.1.0.106               pypi_0    pypi
nvidia-ml-py              12.555.43                pypi_0    pypi
nvidia-nccl-cu11          2.20.5                   pypi_0    pypi
nvidia-nccl-cu12          2.20.5                   pypi_0    pypi
nvidia-nvjitlink-cu12     12.6.20                  pypi_0    pypi
nvidia-nvtx-cu11          11.8.86                  pypi_0    pypi
nvidia-nvtx-cu12          12.1.105                 pypi_0    pypi
oauthlib                  3.2.2                    pypi_0    pypi
openssl                   1.1.1w               h7f8727e_0    defaults
optimum                   1.16.2                   pypi_0    pypi
orjson                    3.10.6                   pypi_0    pypi
packaging                 23.2                     pypi_0    pypi
pandas                    2.2.2                    pypi_0    pypi
peft                      0.12.0                   pypi_0    pypi
pillow                    10.4.0                   pypi_0    pypi
pip                       24.0            py310h06a4308_0    defaults
platformdirs              4.2.2                    pypi_0    pypi
prompt-toolkit            3.0.47                   pypi_0    pypi
proto-plus                1.24.0                   pypi_0    pypi
protobuf                  4.25.4                   pypi_0    pypi
psutil                    6.0.0                    pypi_0    pypi
py-cpuinfo                9.0.0                    pypi_0    pypi
pyarrow                   17.0.0                   pypi_0    pypi
pyarrow-hotfix            0.6                      pypi_0    pypi
pyasn1                    0.6.0                    pypi_0    pypi
pyasn1-modules            0.4.0                    pypi_0    pypi
pydantic                  2.6.3                    pypi_0    pypi
pydantic-core             2.16.3                   pypi_0    pypi
pydub                     0.25.1                   pypi_0    pypi
pygments                  2.18.0                   pypi_0    pypi
pynvml                    11.5.3                   pypi_0    pypi
pyparsing                 3.1.2                    pypi_0    pypi
python                    3.10.0               h12debd9_5    defaults
python-dateutil           2.9.0.post0              pypi_0    pypi
python-dotenv             1.0.1                    pypi_0    pypi
python-multipart          0.0.9                    pypi_0    pypi
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
readline                  8.2                  h5eee18b_0    defaults
referencing               0.35.1                   pypi_0    pypi
regex                     2024.7.24                pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
requests-oauthlib         2.0.0                    pypi_0    pypi
responses                 0.18.0                   pypi_0    pypi
rich                      13.7.1                   pypi_0    pypi
rpds-py                   0.20.0                   pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
s3fs                      2024.3.1                 pypi_0    pypi
safetensors               0.4.4                    pypi_0    pypi
scikit-learn              1.2.2                    pypi_0    pypi
scipy                     1.14.0                   pypi_0    pypi
semantic-version          2.10.0                   pypi_0    pypi
sentencepiece             0.2.0                    pypi_0    pypi
sentry-sdk                2.12.0                   pypi_0    pypi
setproctitle              1.3.3                    pypi_0    pypi
setuptools                72.1.0          py310h06a4308_0    defaults
shortuuid                 1.0.13                   pypi_0    pypi
shtab                     1.7.1                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.1                    pypi_0    pypi
sniffio                   1.3.1                    pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0    defaults
starlette                 0.37.2                   pypi_0    pypi
svgwrite                  1.4.3                    pypi_0    pypi
sympy                     1.13.1                   pypi_0    pypi
tensorboard               2.17.0                   pypi_0    pypi
tensorboard-data-server   0.7.2                    pypi_0    pypi
termcolor                 2.4.0                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tiktoken                  0.7.0                    pypi_0    pypi
tk                        8.6.14               h39e8969_0    defaults
tokenizers                0.19.1                   pypi_0    pypi
toolz                     0.12.1                   pypi_0    pypi
torch                     2.3.1+cu118              pypi_0    pypi
tqdm                      4.66.5                   pypi_0    pypi
transformers              4.43.4                   pypi_0    pypi
triton                    2.3.1                    pypi_0    pypi
trl                       0.9.6                    pypi_0    pypi
typing-extensions         4.12.2                   pypi_0    pypi
tyro                      0.8.5                    pypi_0    pypi
tzdata                    2024.1                   pypi_0    pypi
urllib3                   2.2.2                    pypi_0    pypi
uvicorn                   0.30.5                   pypi_0    pypi
wandb                     0.17.6                   pypi_0    pypi
wavedrom                  2.0.3.post3              pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
websockets                11.0.3                   pypi_0    pypi
werkzeug                  3.0.3                    pypi_0    pypi
wheel                     0.43.0          py310h06a4308_0    defaults
wrapt                     1.16.0                   pypi_0    pypi
xformers                  0.0.27                   pypi_0    pypi
xxhash                    3.4.1                    pypi_0    pypi
xz                        5.4.6                h5eee18b_1    defaults
yarl                      1.9.4                    pypi_0    pypi
zlib                      1.2.13               h5eee18b_1    defaults
zstandard                 0.22.0                   pypi_0    pypi
```

**Comment by Mr-Jeffery** [comment]
I changed to `python==3.10` and everything works

**Comment by griff4692** [comment]
I switched to python 3.10 and still getting same issue. Any solution?

**Comment by Calandiel** [comment]
From the quickstart section: `Requirements: Python >=3.10 and Pytorch >=2.1.1.`

I tried running it with Python 3.12 and got the same error as the OP.

**Comment by chiwanpark** [comment]
One temporary solution is to install AutoAWQ and xformers separately before installing axolotl. Here are commands which I succeeded.

```bash
# install xformers with torch
pip3 install 'torch<=2.4.1' xformers -i https://download.pytorch.org/whl/cu124

# install autoawq
TORCH_VERSION=2.4.1+cu124 INSTALL_KERNELS=1 pip3 install git+https://github.com/casper-hansen/AutoAWQ.git

# install axolotl
pip3 install -e '.[deepspeed,flash-attn,mlflow]' -i https://download.pytorch.org/whl/cu124
```

**Comment by jackswl** [comment]
latest update to fix gradient accumulation can be done with what @chiwanpark mentioned, but removing the '-i https://download.pytorch.org/whl/cu124' for the last part on # install axolotl. This will allow 'trl' to be installed properly.

i have done this using python=3.10, cuda 12.1

**Comment by winglian** [comment]
just to confirm, are you using PyTorch versions >= 2.3.1 ? torch==2.1.1 is no longer supported.

**Comment by bursteratom** [comment]
Closing this issue as following @Mr-Jeffery's steps to reproduce on torch>=2.3.1 no longer produces error.
